### PR TITLE
[Fix] Hide vanished players from server list pings

### LINKED ISSUE #837: Hide vanished players from server list.
Vanished players are visible in the servers list on client.

I mean this:

### PR DESCRIPTION
**Pull request by mdcfe**
Add listener for ServerListPingEvent and remove any vanished players. Fixes #837.
Also refactored EssentialsPluginListener serverListener to pluginListener

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/EssentialsServerListener.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import net.ess3.api.IEssentials;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.server.ServerListPingEvent;
+
+import java.util.Iterator;
+
+
```
Comment (mbax) [inline Essentials/src/com/earth2me/essentials/EssentialsServerListener.java]: Two linebreaks?

**Comment by ghost** [comment]
Was working on a solution to this until I saw this pull request, +1

**Comment by ScuroK** [comment]
players still see me on the tablist if im vanish.

**Comment by mdcfe** [comment]
What server version are you on?

**Comment by ScuroK** [comment]
latest spigot build and latest essentialsx. OP-color is set to 'none'

i dont know why but some players told me they saw me in the list.

**Comment by ScuroK** [comment]
Problem seems gone. sorry

**Pull request merged**